eospac: new default version 6.5.0

KineticTheory opened the pull request:
* Provide eospac@6.5.0
* Mark new version as the default (was 6.4.2)
* New version fixes #17370
* Also includes changes from spack#21629

KineticTheory committed:
[pkg][new version] Provide eospac@6.5.0 and mark it as default.

KineticTheory committed:
Merge remote-tracking branch 'origin/develop' into eospac650default

KineticTheory committed:
Merge in changes found in #21629

sethrj commented on var/spack/repos/builtin/packages/eospac/package.py:
```diff
@@ -21,10 +21,16 @@ class Eospac(Package):
-    version('6.4.2', preferred=True,
+    version('6.5.0alpha.1',
+            sha256='c7334342dd2e071e17c60d8fabb11d2908c9d48c9b49ea83c3609a10b7b8877b',
+            url="http://laws.lanl.gov/projects/data/eos/get_file.php?package=eospac&filename=eospac_v6.5.0alpha.1_62053188b9842842e41508c54196c533e0b91e51.tgz")
```
Mark the alpha and beta versions with `deprecated=True`, which will tell the concretizer to prefer a different version. The version comparisons for spack is weird otherwise:
```
$ spack-python
Spack version 0.16.3
Python 3.8.8, Darwin x86_64
>>> from spack.version import ver
>>> ver('6.5.0alpha.1') > ver('6.5.0')
True
```

KineticTheory replied:
Fixed in 1a25945

KineticTheory committed:
Mark all alpha/beta versions as deprecated.

- Addresses @sethrj's recommendation
- Also add a note indicated why these versions are marked this way.
